### PR TITLE
Substake lock up duration impact on supply, rewrite avoiding inflation rate concept

### PR DESCRIPTION
**Pull request by arjunhassard**
See https://github.com/nucypher/nucypher/blob/0abfd96b463d1d95000ed91e188be58382439ee7/nucypher/blockchain/eth/sol/source/contracts/Issuer.sol#L140
for current calculation of given substake's reward in a period (eq. 9 in paper)

### CODE REVIEW

File: 2_phase_minting_schedule/main.tex [hunk 1]
```diff
@@ -4,61 +4,62 @@
...
+Monetary policy designed for decentralized networks often make reference to a so-called `inflation rate'. In this model, phase one sees an \textbf{identical sum of tokens} minted in each period, so the effective `rate' at which the circulating supply grows decreases period-to-period (and year-to-year). In phase two, the number of tokens minted per day also decreases, which accelerates the decline in the supply growth rate. To avoid misleading terminology, the number of total tokens minted per day shall be hereafter referred to as the \textit{issuance}. Note that the effective growth rate of any given stake also decreases over time, the extent to which is subject to discretionary factors such as the percentage of tokens re-staked by the node operator.
```
Comment (mswilkison) [inline 2_phase_minting_schedule/main.tex]:
```suggestion
Monetary policy designed for decentralized networks often make reference to a so-called `inflation rate'. In this model, phase one sees an \textbf{constant amount of tokens} minted in each period, so the effective `rate' at which the circulating supply grows decreases period-to-period (and year-to-year). In phase two, the number of tokens minted per day also decreases, which accelerates the decline in the supply growth rate. To avoid misleading terminology, the number of total tokens minted per day shall be hereafter referred to as the \textit{issuance}. Note that the effective growth rate of any given stake also decreases over time, the extent to which is subject to discretionary factors such as the percentage of tokens re-staked by the node operator. 
```

File: 2_phase_minting_schedule/main.tex [hunk 2]
```diff
@@ -4,61 +4,62 @@
...
 \section{Motivation \& Rationale}
 
-\textit{Demand-oriented rationale}
-\\\\
-The following premises form the basis for this proposal: 
+\subsection{Demand uncertainty}
+
+These demand-related premises form the basis for the two-phase model: 
 \begin{enumerate}
 \item There is little certainty regarding the length of time that will elapse between the network's launch and the first trickle of demand from customers (developers, end-users \& others).
```
Comment (mswilkison) [inline 2_phase_minting_schedule/main.tex]:
```suggestion
\item There is little certainty regarding the length of time that will elapse between the network's launch and the development of meaningful demand from customers (developers, end-users \& others).
```

File: 2_phase_minting_schedule/main.tex [hunk 3]
```diff
@@ -4,61 +4,62 @@
...
-\item At least some early customers will inflexibly require long-lasting sharing policies. This means that the protocol must motivate the successive re-commitment to long token lock durations (i.e. staking for 12+ months) by a sufficient population of node operators. A diminished inflation rate, in lieu of a mature fee market, may not be enough to incentivize this repeated and essential re-commitment. 
+\item A popular subsidy model involves daily token issuance dropping exponentially, right from launch. If the fee market does not sufficiently mature in time to make up for the sharp reduction in earnings the early years of the network's existence, some node operations may be rendered entirely unsustainable. 
+\item Most importantly, some of NuCypher's early customers will (inflexibly) require long-lasting sharing policies. This means that the protocol must motivate the successive re-commitment to long token lock durations (i.e. staking for 12+ months) by a sufficient population of node operators. A diminished subsidy issuance, in lieu of a mature fee market, may not be enough to incentivize this repeated and essential re-commitment.
```
Comment (mswilkison) [inline 2_phase_minting_schedule/main.tex]:
```suggestion
\item Most importantly, some of NuCypher's early customers may (inflexibly) require long-lasting sharing policies. This means that the protocol must motivate the successive re-commitment to long token lock durations (i.e. staking for 12+ months) by a sufficient population of node operators. A diminished subsidy issuance, in lieu of a mature fee market, may not be enough to incentivize this repeated and essential re-commitment. 
```

File: 2_phase_minting_schedule/main.tex [hunk 4]
```diff
@@ -4,61 +4,62 @@
...
 \\\\
-As mentioned, the two-phase model implies a lower genesis inflation rate than the previous draft model. This itself confers further advantages to network health:  
+Hence, a two-phase minting schedule assists all would-be node operators who don't happen to stake right at network launch, but decide to do so at a later date. By addressing the temporal inequity of the previous draft, we impede problematic centralization trends.
```
Comment (mswilkison) [inline 2_phase_minting_schedule/main.tex]:
```suggestion
Hence, a two-phase minting schedule assists all would-be node operators who don't happen to stake right at network launch, but decide to do so at a later date. By addressing the temporal inequity of immediately decreasing issuance, we impede problematic centralization trends.
```

File: 2_phase_minting_schedule/main.tex [hunk 5]
```diff
@@ -4,61 +4,62 @@
...
+
+\subsection{Risks to operational feasibility, incentives and pricing}
+
+As mentioned, the two-phase model implies, relative to other subsidy models, a lower number of tokens minted on day one. This in itself confers further advantages to network health:
```
Comment (mswilkison) [inline 2_phase_minting_schedule/main.tex]: This is not really clear since even with two-phase inflation, the amount of tokens minted on day one is the highest it will ever be.

**Review by mswilkison — APPROVED** [review]

File: 2_phase_minting_schedule/main.tex (hunk 5 above)
Comment (arjunhassard) [reply 2_phase_minting_schedule/main.tex]:
```suggestion
As Figure 1 shows, a `generic' subsidy model involves high issuance at genesis. In the two phase model, the number of tokens minted per day isn't only stable in phase one, it also commences and remains a lower figure for the first few years. This difference confers further benefits to network health:  
```
Agree, the use of 'day one' is unnecessary.

File: 2_phase_minting_schedule/main.tex [hunk 6]
```diff
@@ -4,61 +4,62 @@
...
```
Comment (arjunhassard) [inline 2_phase_minting_schedule/main.tex]:
```suggestion
As Figure 1 shows, a `generic' subsidy model involves high issuance at genesis. In the two phase model, the number of tokens minted per day isn't only stable in phase one, it also commences and remains a lower figure for the first few years. This difference confers further benefits to network health:  
```
Agreed, the use of 'day one' is unnecessary. @mswilkison

File: 2_phase_minting_schedule/main.tex [hunk 7]
```diff
@@ -70,74 +71,103 @@ \section{Empirical analysis of existing networks}
...
 \\\\
-To avoid confusing the inflation rate with the real growth rate of the supply, we hereafter suggest the term \textit{minting schedule} to describe the growth of the circulating supply.
+In this token minting schedule, we can work out that approximately 1.83 billion tokens will be minted in the first phase, followed by a further 1.056 billion between the commencement of phase two and t = $\infty$.
+
+\subsection{Variable stake lock-up duration}
+
+In the NuCypher protocol, the number of tokens distributed to node operators in a given period depends on the stage in the minting schedule we are at and the size of their stake, but is also a function of the remaining duration their stake is locked up for (in fact, the duration of each of the `sub-stakes' they control). Any sub-stake locked up for a short duration receives less than the maximum subsidy, thereby slowing the expansion of the entire circulating supply – tokens are only minted if they are to be distributed to node operators. The number of tokens minted daily for a given sub-stake depends on the coefficient $\kappa$:
```
Comment (derekpierre) [inline 2_phase_minting_schedule/main.tex]: "depends on the [current] stage in the minting schedule and..."
"also a function of the remaining [locked duration of their stake] (in fact..."
"Any sub-stake [locked] for a ..."

```suggestion
In the NuCypher protocol, the number of tokens distributed to node operators in a given period depends on the current stage in the minting schedule and the size of their stake, but is also a function of the remaining locked duration of their stake (in fact, the duration of each of the `sub-stakes' they control). Any sub-stake locked for a short duration receives less than the maximum subsidy, thereby slowing the expansion of the entire circulating supply – tokens are only minted if they are to be distributed to node operators. The number of tokens minted daily for a given sub-stake depends on the coefficient $\kappa$:
```

File: 2_phase_minting_schedule/main.tex [hunk 8]
```diff
@@ -70,74 +71,103 @@ \section{Empirical analysis of existing networks}
...
-With this new token minting schedule, we can work out that approximately 1.83 billion tokens are minted in the first 5 years, followed by approximately a further 1.056 billion between the fifth year and t = $\infty$.
+For reference, the supply in the second phase can be calculated with this equation: 
 
-\end{document}
+\begin{equation}
+    S(t) = S_0 \left[1 + \frac{I_{stable} \cdot T_{1/2}}{\ln{2}}\left(1 - 2^{-\frac{t \cdot \kappa^*}{T_{1/2}}} \right) \right].
+\end{equation}
 
-In other words, the same amount of tokens will be emitted from $t=5$ until $t=\infty$ 
+Which implies that the rate of issuance decay is inversely proportional to $\kappa^*$. In a similar fashion to phase one, shorter average lock-ups slow down the minting of tokens.
```
Comment (derekpierre) [inline 2_phase_minting_schedule/main.tex]: Is the inverse proportional relationship of "rate" & "kappa" correct here? We should be careful using the word "rate" here.
If kappa is low, i.e. short durations, then the decay rate slows i.e. lowers (and vice versa)

Do you instead mean that the **half-life** (i.e. the amount of time for the quantity to decrease by half) is inversely proportional to kappa?

**Review by derekpierre — COMMENTED** [review]
Great work @arjunhassard!

File: 2_phase_minting_schedule/main.tex (hunk 3 above)
Comment (derekpierre) [reply 2_phase_minting_schedule/main.tex]: ^ Ignore above, it was an old change, before you made updates based on maclane's suggestions.

File: 2_phase_minting_schedule/main.tex (hunk 8 above)
Comment (arjunhassard) [reply 2_phase_minting_schedule/main.tex]: Well-spotted! It's proportional, and I agree about 'rate' (the term that shall not be named). 

The reason I didnt say that kappa* is inversely proportional to the 'half-life' is that we previously say the half-life _is T_1/2 _. Kappa doesn't affect T_1/2  – as you can see from eq. 13, it modifies the expression for decay over time as a whole: -2^(-t * k / T_1/2) 

```suggestion
Which implies that issuance decay is proportional to $\kappa^*$. In a similar fashion to phase one, shorter average lock-ups slow down the minting of tokens. 
```

File: 2_phase_minting_schedule/main.tex [hunk 9]
```diff
@@ -4,61 +4,62 @@
...
+\item The greater the real-world value of the subsidy, the further from a fee-driven `reality' a node's set-up may be in economic terms (for example, in terms of operational efficiency). When subsidies eventually run out, node operators who are unable to adapt may cease operations, threatening general service availability or even reneging on existing commitments to users. Under a model with higher token issuance early on, these inefficient operators may control a substantial proportion of the circulating supply by this point. 
+\item If subsidies are worth significantly more than the typical sum of transaction fees earned in the same period, then the incentive to earn those fees may be blunted. This is particularly problematic for the protocol if the node operator duty required to collect subsidies does not align well with the duty required to earn fees, and the reliability of the latter impacts overall service quality.
```
[the remark this answers is not in the document]
Comment (arjunhassard) [reply 2_phase_minting_schedule/main.tex]: Duty does sound a little pompous! Work is fine, it just doesn't strongly emphasise the distinction between various staker responsibilities, which is the key point here

File: 2_phase_minting_schedule/main.tex (hunk 6 above)
Comment (arjunhassard) [reply 2_phase_minting_schedule/main.tex]: I think what I was originally trying to convey here wasn't v clear
```suggestion
As illustrated in Figure 1, `generic' subsidy models tend to involve a high issuance at genesis and in the first few years of the network's existence. Setting the peak issuance (the maximum number of tokens minted per day) at a comparatively lower figure addresses the following potential problems: 
```

Comment (derekpierre) [reply 2_phase_minting_schedule/main.tex:50]: 👍

File: 2_phase_minting_schedule/main.tex (hunk 8 above)
Comment (derekpierre) [reply 2_phase_minting_schedule/main.tex]: 👍 - Yes decay and not T_1/2 - I misread "8 1/3 years" and "5 years \& 79 days".

**Review by derekpierre — APPROVED** [review]
🎸
